src/app/script/canvas_widget.cpp

```diff
@@ -99,6 +119,11 @@ bool Canvas::onProcessMessage(ui::Message* msg)
 
     case ui::kMouseMoveMessage: {
       auto mouseMsg = static_cast<ui::MouseMessage*>(msg);
+      std::unique_ptr<ui::MouseMessage> scaledMouseMsg;
+      if (m_autoScaling) {
+        scaledMouseMsg = makeScaled(mouseMsg, bounds().origin());
+        mouseMsg = scaledMouseMsg.get();
+      }
       MouseMove(mouseMsg);
```

Comment:
We can change this to (with the new makeScaled()):
```suggestion
      auto mouseMsg = *static_cast<ui::MouseMessage*>(msg);
      if (m_autoScaling)
        mouseMsg = makeScaled(&mouseMsg, bounds().origin());
      MouseMove(&mouseMsg);
```
I think it's better to use values directly (instead of shared pointers here).